scvi/dataset/anndata.py

```diff
@@ -79,13 +80,17 @@ def extract_data_from_anndata(self, ad: anndata.AnnData):
             ad.obs
         )  # provide access to observation annotations from the underlying AnnData object.
 
+        # treat all possible cases according to anndata doc
         if isinstance(ad.X, np.ndarray):
-            data = ad.X.copy()  # Dense
-        else:
-            data = ad.X.toarray()  # Sparse
-
-        select = data.sum(axis=1) > 0  # Take out cells that doesn't express any gene
-        data = data[select, :]
+            data = ad.X.copy()
+        if isinstance(ad.X, pd.DataFrame):
+            data = ad.X.values
+        if isinstance(ad.X, csr_matrix):
+            # keep sparsity above 1 Gb in dense form
+            if ad.X.toarray().nbytes < 1e9:
```

Comment:
Does this call convert the sparse matrix to dense to find out how many bytes it would use if converted?

Reply:
Could you multiply the number of genes by the number of cells by 4 (for single precision floats) to estimate the size the dense matrix would be, without actually forming the dense matrix?